t/document/file.t

```diff
@@ -212,6 +212,13 @@ subtest 'Packages starting with underscore are not indexed' => sub {
     is( $file->module->[0]->indexed, 0, 'Package is not indexed' );
 };
 
+subtest 'files listed under other files' => sub {
+    my $file = new_file_doc( name => 'Makefile.PL' );
+    $file->set_authorized( $file->module => $file->author );
```

Comment:
Oh, after reading the pod for the sub I see that you were perhaps intending to emulate the `$perms` argument
in which case there are different problems:
`$file->module` is an array ref (see the previous test) not a module name, and the method accepts a hash **ref** so you'd need curly braces around the module/author pair, something like
`( { $file->module->[0]->{name} => $file->author } )`.
Problem there is that these "other" files are unlikely to ever have modules.

I think passing an empty hash ref would suffice: `( {} )`.
That way if the code goes past the "other files" check it won't error on not getting a hashref.
Files that are not "other files" should still get authorized then (which would be what you want in this case).